### DISCUSSION TITLE
how to eliminate white space to pass it to another api

### DISCUSSION THREAD

**Discussion by Razi321** [fastapi/fastapi#7191] [Questions] [answered]
I'm using fastapi framework to create a wrapper for shopify products api , the goal in to search for a specific product by name so a way to use it directly is : 

 

       GET https://api_key:password@shopname.myshopify.com/admin/products.json?title=here_goes_the_title 

this works fine but when i try to use it in the wrapper like this :

    @app.get('/products?name={title}')
    def get_searched_products(title : str):
            
            r = requests.get(
            "https://api_key:password@shopname.myshopify.com/admin/products.json?title={title}&fields=id,title")
            
            return r.json()

i got : 

    {"detail":"Not Found"}

the product name that i'm using is : **2020 a/w vest knit**

so in the first case ( the case when it works ) : it will be like this : **2020%20a/w%20vest%20knit**

in my opinion i think that's the problem , when my api pass it to shopify api , the shopify api search for it as **2020%20a/w%20vest%20knit** *not* **2020 a/w vest knit** and that's why I'm getting the error not found !

**Comment by includeamin** [comment, marked as answer]
your definition of query is wrong.
try this
``` python
@app.get('/products')
def get_searched_products(title: str):
    print(title)
    r = requests.get(
        "https://api_key:password@shopname.myshopify.com/admin/products.json?title={title}&fields=id,title")

    return r.json()
```
``` sh
curl http://0.0.0.0:PORT/products?title=2020%20a%2Fw%20vest%20knit
```

**Comment by tiangolo** [comment]
Thanks for the help here @includeamin ! :clap: :bow: 

Thanks for reporting back and closing the issue @Razi321  :+1:
